File: app/src/main/java/com/jawnnypoo/geotune/misc/CheckRegistrationReceiver.java
package com.jawnnypoo.geotune.misc;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.database.Cursor;

import com.jawnnypoo.geotune.data.GeoTune;
import com.jawnnypoo.geotune.data.GeoTuneContentProvider;
import com.jawnnypoo.geotune.service.GeoTuneModService;

import java.util.ArrayList;

/**
 * Need your boot receiver to restore the Geofences if you want them to
 * persist across device reboots
 */
public class CheckRegistrationReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        restoreGeofences(context);
    }

    private void restoreGeofences(Context context) {
        //TODO restore geofences after boot
        Cursor cursor = context.getContentResolver().query(GeoTuneContentProvider.CONTENT_URI,
                null, null, null, null);
        ArrayList<GeoTune> geoTunes = new ArrayList<>();
        if (cursor != null) {
            //Iterate on all the entries we got back
            while (cursor.moveToNext()) {
                geoTunes.add(GeoTune.fromCursor(cursor));
            }
            if (!cursor.isClosed()) {
                cursor.close();
            }
        }
        if (geoTunes.size() > 0) {
            GeoTuneModService.reregisterGeoTunes(context, geoTunes);
        }
    }
}

File: app/src/main/java/com/jawnnypoo/geotune/util/CheckRegistrationReceiver.kt
package com.jawnnypoo.geotune.util

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.jawnnypoo.geotune.data.GeoTune
import com.jawnnypoo.geotune.data.GeoTuneContentProvider
import com.jawnnypoo.geotune.service.GeoTuneModService
import java.util.*

/**
 * Need your boot receiver to restore the Geofences if you want them to
 * persist across device reboots
 */
class CheckRegistrationReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        restoreGeofences(context)
    }

    private fun restoreGeofences(context: Context) {
        val cursor = context.contentResolver.query(GeoTuneContentProvider.CONTENT_URI, null, null, null, null)
        val geoTunes = ArrayList<GeoTune>()
        if (cursor != null) {
            //Iterate on all the entries we got back
            while (cursor.moveToNext()) {
                geoTunes.add(GeoTune.fromCursor(cursor))
            }
            if (!cursor.isClosed) {
                cursor.close()
            }
        }
        if (geoTunes.size > 0) {
            GeoTuneModService.reregisterGeoTunes(context, geoTunes)
        }
    }
}